src/main/java/com/mraof/minestuck/computer/editmode/EditToolDrag.java

```diff
@@ -0,0 +1,438 @@
+
+import java.util.UUID;
+
+public class EditToolDrag
+{
+	
+	public static void doReviseCode(TickEvent.ClientTickEvent event)
+	{
+		Minecraft mc = Minecraft.getInstance();
+		if (mc.player == null || event.phase == TickEvent.Phase.END)
+			return;
+		
+		Player player = mc.player;
+		IEditTools cap = player.getCapability(MSCapabilities.EDIT_TOOLS_CAPABILITY, null).orElse(new EditTools());
+		
+		if (cap.getToolMode() != null && cap.getToolMode() != IEditTools.ToolMode.REVISE)
+			return;
+		
+		boolean isDragging = cap.isEditDragging();
+		boolean isDown = mc.options.keyUse.isDown();
+		
+		if (isDown)
+		{
+			
+			if(!canEditRevise(player))
+			{
+				cap.setToolMode(null);
+				cap.setEditDragging(false);
+				cap.setEditPos1(null);
+				cap.setEditPos2(null);
+				return;
+			}
+		
+			if (!isDragging)
+			{
+				BlockHitResult blockHit = getPlayerPOVHitResult(player.getLevel(), player);
+				if (blockHit.getType() == BlockHitResult.Type.BLOCK)
+				{
+					cap.setToolMode(IEditTools.ToolMode.REVISE);
+					
+					cap.setEditPos1(player.level.getBlockState(blockHit.getBlockPos()).getMaterial().isReplaceable() ? blockHit.getBlockPos() : blockHit.getBlockPos().offset(blockHit.getDirection().getNormal()));
+					cap.setEditTraceHit(blockHit.getLocation());
+					cap.setEditTraceDirection(blockHit.getDirection());
+					cap.setEditReachDistance(Math.sqrt(cap.getEditPos1().distToLowCornerSqr(player.getEyePosition().x, player.getEyePosition().y, player.getEyePosition().z)));
+					
+				}
+			}
+			
+			if (cap.getEditPos1() != null)
+			{
+				
+				BlockHitResult blockHit = getPlayerPOVHitResult(player.getLevel(), player);
+				BlockPos pos2;
+				//if not looking directly at a block, use the position where the player is looking at with the initial distance of editPos1 from the camera
+				if (blockHit.getType() == BlockHitResult.Type.MISS)
+				{
+					Vec3 eyePosition = player.getEyePosition();
+					Vec3 lookDirection = player.getLookAngle();
+					Vec3 selectionPosition = eyePosition.add(lookDirection.x * cap.getEditReachDistance(), lookDirection.y * cap.getEditReachDistance(), lookDirection.z * cap.getEditReachDistance());
+					pos2 = new BlockPos(selectionPosition.x, selectionPosition.y, selectionPosition.z);
+				}
+				else
+					pos2 = player.level.getBlockState(blockHit.getBlockPos()).getMaterial().isReplaceable() ? blockHit.getBlockPos() : blockHit.getBlockPos().offset(blockHit.getDirection().getNormal());
+				
+				cap.setEditPos2(pos2);
+				MSPacketHandler.sendToServer(new EditmodeFillPacket(true, isDown, cap.getEditPos1(), cap.getEditPos2(), cap.getEditTraceHit(), cap.getEditTraceDirection()));
+			}
+		}
+		else if (isDragging)
+		{
+			if (cap.getEditPos1() != null)
+			{
+				MSPacketHandler.sendToServer(new EditmodeFillPacket(true, isDown, cap.getEditPos1(), cap.getEditPos2(), cap.getEditTraceHit(), cap.getEditTraceDirection()));
+			}
+			
+			cap.setToolMode(null);
+			cap.setEditPos1(null);
+			cap.setEditPos2(null);
```

Comment:
Seeing calls like these in a bunch of places. How about making a clear function for better readability?